nuitka/OptionParsing.py

```diff
@@ -952,6 +952,17 @@
 and "auto" (when it's known to work). Defaults to "auto".""",
 )
 
+c_compiler_group.add_option(
+    "--fcf-protection",
+    action="store",
+    dest="fcf_protection",
+    metavar="choice",
+    default="full",
+    choices=("full", "branch", "return", "none", "check"),
+    help="""\
+Choice fcf-protection mode""",
```

Comment:
Enabling by default conflicts with performance goals of course, I would use `none` there. This most definitely needs a link to the GCC documentation for docs I guess.